lib/llm/src/block_manager/block/registry.rs

```diff
@@ -82,21 +83,34 @@ impl BlockRegistry {
                     }
                 }
 
-                // Create the [RegistrationHandle] and [PublishHandle]
-                let publish_handle =
-                    Self::create_publish_handle(state.token_block(), self.event_manager.clone());
-                let reg_handle = publish_handle.remove_handle();
+                let mut publish_handle = None;
+
+                let block_handle = Arc::new(());
+
+                let reg_handle = if let Some(existing_handle) = existing_handle {
+                    existing_handle.clone()
+                } else {
```

Comment:
I think there is one condition that need to be considered.  if each pool has this registry with storage-specific-block-handles, it means two pools could simultaneously register a block with the same sequence hash and result in two publish handles and two unique registration handles.

we might need to have a common `Arc<Mutex<Map<SequenceHash, Weak<RegistrationHandle>>>>` which is cloned in each pool's block registry to ensure in the `None` scenario for an existing registration handle, that only one can ever be created across all pools.